bvm/ballerina-core/src/main/java/org/ballerinalang/bre/bvm/BLangVMStructs.java

```diff
@@ -55,15 +56,17 @@
         BStructureType structType = structInfo.getType();
         BMap<String, BValue> bStruct = new BMap<>(structType);
 
-        BField[] structFields = structType.getFields();
-        for (int i = 0; i < structFields.length; i++) {
+        Map<String, BField> structFields = structType.getFields();
+        int valCount = 0;
+        for (Map.Entry<String, BField> entry : structFields.entrySet()) {
```

Comment:
Since we are only using map values for the operation. Shall we only get the values of the map (rather getting entries)?

Reply:
+1. Addressed with commit 979331f